flink-end-to-end-tests/flink-end-to-end-tests-common-kafka/src/test/java/org/apache/flink/tests/util/kafka/SQLClientKafkaITCase.java

```diff
@@ -106,11 +112,16 @@ public SQLClientKafkaITCase(String kafkaVersion, String kafkaSQLVersion, String
 	}
 
 	@Before
-	public void before() {
+	public void before() throws Exception {
+		downloadCache.before();
 		Path tmpPath = tmp.getRoot().toPath();
 		LOG.info("The current temporary path: {}", tmpPath);
 		this.sqlClientSessionConf = tmpPath.resolve("sql-client-session.conf");
 		this.result = tmpPath.resolve("result");
+
+		apacheAvroJars.add(downloadCache.getOrDownload("https://repo1.maven.org/maven2/org/apache/avro/avro/1.8.2/avro-1.8.2.jar", tmpPath));
```

Comment:
So this only was only working by chance since missing stuff was provided by flink-shaded-hadoop? If so, why does that no longer work?
Related (and likely the correct fix): https://issues.apache.org/jira/browse/FLINK-17417

Reply:
> So this only was only working by chance since missing stuff was provided by flink-shaded-hadoop? 

Yes.

> If so, why does that no longer work?

Because flink-shaded-hadoop does not exist anymore in this change

Re correct fix: I discussed this offline with @dawidwys and @twalthr and they agreed that providing the required avro dependencies is an acceptable approach.

Reply:
> Because flink-shaded-hadoop does not exist anymore in this change

Well duh, but hadoop is still on the classpath, no? Otherwise you're not fulfilling the contract of the `Hadoop` category; if the test is still passing then the category should be removed.